### PR TITLE
[SDK-563] Add Environment tests

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This adds test for `Environment`, which was untested so far.

## Motivation and Context

In order to improve the testsbase this PR was made.
This is part of  [SDK-563](https://gesundheitscloud.atlassian.net/browse/SDK-563) and depends on #124.

## How is it being implemented?

It makes a minor code changes to the Environment in terms to make the code looking nicer.

## How Has This Been Tested?

The main purpose of the PR is to add the tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sdk-core/src/main/java/care/data4life/sdk/network/Environment.kt [hunk 1]
```diff
@@ -15,79 +15,71 @@
...
+            else -> throw IllegalArgumentException("No supported platform found for value($platform)")
         }
-        throw IllegalArgumentException("No supported platform found for value($platform)")
     }
 
     private fun d4lCertificatePin(): String {
         return when (this) {
-            SANDBOX, DEVELOPMENT, LOCAL -> "sha256/3f81qEv2rjHvcrwof2egbKo5MjjSHaN/4DOl7R+pH0E="
-            STAGING, PRODUCTION -> "sha256/AJvjswWs1n4m1KDmFNnTqBit2RHFvXsrVU3Uhxcoe4Y="
+            SANDBOX, DEVELOPMENT, LOCAL -> D4L_SDL_CERT
```
Comment (wmontwe) [inline sdk-core/src/main/java/care/data4life/sdk/network/Environment.kt:57]: the certificates are bound to a domain, I would reflect that in the name: 
* D4L_SDL_CERT is hpsgc.de
* D4L_SP_CERT is data4life.care
* S4H_CERT is smart4health.eu

**Review by wmontwe — APPROVED** [review]
👍

File: sdk-core/src/main/java/care/data4life/sdk/network/Environment.kt (hunk 1 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/network/Environment.kt:57]: Good point, so more using the domain instead a generic D4l_SOMETHING_CERT name?

Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/network/Environment.kt]: done

**Review by wmontwe — DISMISSED** [review]
👍

**Review by wmontwe — APPROVED** [review]

**Pull request merged**